File: bidappdemoapp-java/src/main/java/io/bidapp/demo/MainActivity.java
package io.bidapp.demo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;



public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        TextView interstitialButton = findViewById(R.id.inter);
        TextView rewardedButton = findViewById(R.id.reward);
        TextView bannersButton = findViewById(R.id.banners);
        ConstraintLayout bannerMain = findViewById(R.id.bannerMain);

        FullscreenLoadDelegate loadDelegate = new FullscreenLoadDelegate(this);
        Interstitial.INSTANCE.loadDelegate = loadDelegate;
        Rewarded.INSTANCE.loadDelegate = loadDelegate;

        FullscreenShowDelegate showDelegate = new FullscreenShowDelegate(this);
        BannerViewDelegate bannerShowDelegate = new BannerViewDelegate(this);
        BannerView banner = new BannerView(this).banner(bannerShowDelegate, AdFormat.banner_320x50);

        BIDConfiguration bidConfig = new BIDConfiguration();


        bidConfig.enableTestMode();

        bidConfig.enableTestMode();
        bidConfig.enableLogging();
        bidConfig.enableInterstitialAds();
        bidConfig.enableRewardedAds();
        bidConfig.bannerEnabled_300x250();
        bidConfig.bannerEnabled_320x50();


        String pubid = "15ddd248-7acc-46ce-a6fd-e6f6543d22cd";

        BidappAds.start(this, pubid, bidConfig);

        interstitialButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Interstitial.INSTANCE.show(showDelegate);
            }
        });

        rewardedButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Rewarded.INSTANCE.show(showDelegate);
            }
        });

        bannersButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), BannerActivity.class);
                startActivity(intent);
            }
        });
        bannerMain.addView(banner);
    }
}

File: bidappdemoapp-compose/app/src/main/java/com/bidapp/demo/MainActivity.kt
package com.bidapp.demo

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.FrameLayout
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.DisposableEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.viewinterop.AndroidView
import io.bidapp.sdk.AdFormat
import io.bidapp.sdk.BIDConfiguration
import io.bidapp.sdk.BannerView
import io.bidapp.sdk.BidappAds
import io.bidapp.sdk.Interstitial
import io.bidapp.sdk.Rewarded
import kotlinx.coroutines.Runnable

class MainActivity : ComponentActivity() {
    private val handler = Handler(Looper.getMainLooper())
    private var runnable : Runnable? = null
    lateinit var interstitial: Interstitial
    lateinit var rewarded: Rewarded
    val loadDelegate = FullscreenLoadDelegate()
    val showDelegate = FullscreenShowDelegate()
    val bannerShowDelegate = BannerViewDelegate()
    var banner: BannerView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val config = BIDConfiguration()
        config.enableLogging()
        config.enableTestMode()
        config.enableInterstitialAds()
        config.enableRewardedAds()
        config.enableBannerAds_300x250()
        config.enableBannerAds_320x50()

        interstitial = Interstitial(this)
        interstitial.setLoadDelegate(loadDelegate)

        rewarded = Rewarded(this)
        rewarded.setLoadDelegate(loadDelegate)

        banner = addBanner_320x50(bannerShowDelegate)



        val pubid = "15ddd248-7acc-46ce-a6fd-e6f6543d22cd"
        BidappAds.start(pubid, config, this)

        setContent {
              Main()
        }
    }

    @Composable
    fun Main(){
        val context = LocalContext.current
        DisposableEffect(Unit) {
            onDispose {
                runnable?.let { handler.removeCallbacks(it) }
            }
        }

        Column(
            modifier = Modifier
                .height(300.dp)
                .fillMaxSize(),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            TextButton(text = "Show Interstitial") {
                interstitial.showWithDelegate(this@MainActivity, showDelegate)
            }
            TextButton(text = "Show Rewarded") {
                rewarded.showWithDelegate(this@MainActivity, showDelegate)
            }
            TextButton(text = "Show Banners") {
                val intent = Intent(context, BannersActivity::class.java)
                context.startActivity(intent)
            }

        }
        Column(
            modifier = Modifier.fillMaxSize(),
            verticalArrangement = Arrangement.Bottom,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Box(
                modifier = Modifier
                    .width(320.dp)
                    .fillMaxWidth()
            ) {
                AndroidView(
                    factory = {
                        val view = FrameLayout(this@MainActivity)
                        view.removeAllViews()
                        view.addView(banner)
                        view
                    },
                    modifier = Modifier.matchParentSize()
                )
            }
        }
    }
    @Composable
    fun TextButton(text: String, onAction: () -> Unit) {
        var color by remember { mutableStateOf(Color.Blue) }
        Text(
            text = text,
            color = color,
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            modifier = Modifier
                .padding(0.dp, 8.dp)
                .clickable {
                    runnable = Runnable {
                        color = Color.Blue
                        onAction()
                    }
                    color = Color.Red
                    handler.postDelayed(runnable!!, 100)
                }



        )

    }
    fun addBanner_320x50(bannerShowDelegate: BannerViewDelegate): BannerView {
        val banner = BannerView(this).banner(AdFormat.banner_320x50)
        banner.setBannerViewDelegate(bannerShowDelegate)
        return banner
    }

}






